SocketRocket/SRWebSocket.m

```diff
@@ -1403,90 +1444,98 @@ - (void)stream:(NSStream *)aStream handleEvent:(NSStreamEvent)eventCode;
     }
 
     dispatch_async(_workQueue, ^{
-        switch (eventCode) {
-            case NSStreamEventOpenCompleted: {
-                SRFastLog(@"NSStreamEventOpenCompleted %@", aStream);
-                if (self.readyState >= SR_CLOSING) {
-                    return;
-                }
-                assert(_readBuffer);
-                
-                if (self.readyState == SR_CONNECTING && aStream == _inputStream) {
-                    [self didConnect];
-                }
-                [self _pumpWriting];
-                [self _pumpScanner];
-                break;
+        [weakSelf safeHandleEvent:eventCode stream:aStream];
+    });
+}
+
+- (void)safeHandleEvent:(NSStreamEvent)eventCode stream:(NSStream *)aStream
+{
+    __strong typeof(self) strongSelf = self;
+    ((void)strongSelf);
```

Comment:
You're right Mike. Pre-arc school says 'that's not accurate', but post arc school says otherwise. Thank you, updating =)